src/index.spec.ts

```diff
@@ -84,11 +83,59 @@ describe('fetch-with-token', () => {
       ]
     `);
 
-    server.service.on('beforeIntrospect', (introspectResponse) => {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    function mockIntrospect(introspectResponse: any) {
       introspectResponse.statusCode = 401;
       introspectResponse.body = { active: false };
+    }
+
+    server.service.on('beforeIntrospect', mockIntrospect);
+    await expect(introspect(), 'should not retry infinitely').resolves.toEqual({
+      status: 401,
+      json: { active: false },
     });
-    await expect(introspect(), 'should not retry infinitely').resolves.toEqual({ status: 401, json: { active: false } });
+    server.service.off('beforeIntrospect', mockIntrospect);
```

Comment:
This listener was bleeding out from the test, so removing it after the assertion.